website/source/docs/audit/socket.html.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: "docs"
+page_title: "Audit Backend: Socket"
+sidebar_current: "docs-audit-socket"
+description: |-
+  The "socket" audit backend writes audit writes to a TCP or UDP socket.
+---
+
+# Audit Backend: Socket
+
+The `socket` audit backend writes to a TCP or UDP socket.
+
+## Format
+
+Each line in the audit log is a JSON object. The `type` field specifies what type of
+object it is. Currently, only two types exist: `request` and `response`. The line contains
+all of the information for any given request and response. By default, all the sensitive
+information is first hashed before logging in the audit logs.
+
+## Enabling
+
+#### Via the CLI
+
+Audit `socket` backend can be enabled by the following command.
+
+```
+$ vault audit-enable socket
+```
+
+Backend configuration options can also be provided from command-line.
+
+```
+$ vault audit-enable socket address="127.0.0.1:9090" socket_type="tcp"
+```
+
+Following are the configuration options available for the backend.
+
+<dl class="api">
+  <dt>Backend configuration options</dt>
+  <dd>
+    <ul>
+      <li>
+        <span class="param">address</span>
+        <span class="param-flags">optional</span>
+            The socket server address to use. Defaults to `127.0.0.1:9090`.
+      </li>
+      <li>
+        <span class="param">socket_type</span>
+        <span class="param-flags">optional</span>
+            The socket type to use (either `tcp` or `udp`). Defaults to `tcp`.
```

Comment:
I was going to ask for unix socket support, but I _think_ this PR is already compatible with unix sockets.

Can we have this say something along the lines of "any socket type understood by net.Dial is acceptable, including `tcp` and `udp`".